Bottom: 5a8683d41e60a6727e48248f8a768cc6cd9aeefc
Top:    d0cd5c7a6955498affe96b573ff0e0528adc07f6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-23 18:21:24 +0000

Workaround crash where contour is passed as NULL to poly_PreContour







---

diff --git a/src/polygon.c b/src/polygon.c
index 64cc662..5f6893d 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -202,6 +202,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
